test/raft/randomized_nemesis_test.cc

```diff
@@ -1381,6 +1381,11 @@ class raft_server {
         auto fd_ep_map = std::make_unique<direct_fd_endpoint_map>();
         auto fd_service = std::make_unique<sharded<direct_failure_detector::failure_detector>>();
         auto update_fd_server = [&fd = *fd_service, &ep_map = *fd_ep_map] (raft::server_id id, bool added) {
+            if (!fd.local_is_initialized()) {
```

Comment:
e.g. `stop_crash` operation. It aborts servers. Aborting a server stops the failure detector among other things.

Reply:
Shouldn't we first stop the server, then fd?

Reply:
```cpp
            // Stop _fd_service before _server because _fd_service is using _server's RPC.
            // _server may try to add/remove endpoints after _fd_service is stopped but it's allowed.
            _fd_subscription = std::nullopt;
            co_await _fd_service->stop();
            co_await _server->abort();
```

`fd` is using `server`'s RPC.
It would be a bit of pain to untangle the dependency loop.